librz/core/cmd/cmd_search_rop.c

```diff
@@ -4,726 +4,402 @@
-			ops_list = parse_list(strstr(out, "ops.list"));
-			flg_read = parse_list(strstr(out, "flg.read"));
-			flg_write = parse_list(strstr(out, "flg.write"));
-			reg_read = parse_list(strstr(out, "reg.read"));
-			reg_write = parse_list(strstr(out, "reg.write"));
-			mem_read = parse_list(strstr(out, "mem.read"));
-			mem_write = parse_list(strstr(out, "mem.write"));
-		} else {
-			goto continue_error;
-		}
+	if (!parse_il_equal(str, &idx)) {
+		free(dst_reg);
+		return false;
+	}
 
-		if (!rz_list_find(ops_list, "=", (RzListComparator)strcmp, NULL)) {
-			goto continue_error;
-		}
+	char *src_reg = parse_register(core, str, &idx);
+	if (!src_reg) {
+		free(dst_reg);
+		return false;
+	}
+	RzList *args = rz_list_new();
+	if (!parse_il_op(args, str, &idx)) {
+		free(dst_reg);
+		free(src_reg);
+		rz_list_free(args);
+		return false;
+	}
 
-		head = rz_reg_get_list(core->analysis->reg, RZ_REG_TYPE_GPR);
-		if (!head) {
-			goto out_error;
-		}
-		rz_list_foreach (head, iter_dst, item_dst) {
-			ut64 diff_dst, value_dst;
-			if (!rz_list_find(reg_write, item_dst->name,
-				    (RzListComparator)strcmp, NULL)) {
-				continue;
-			}
+	ut64 const_value;
+	if (!parse_constant(str, &idx, &const_value)) {
+		free(dst_reg);
+		free(src_reg);
+		return false;
+	}
 
-			// you never mov into flags
-			if (isFlag(item_dst)) {
-				continue;
-			}
+	if (!parse_eof(str, idx)) {
+		free(dst_reg);
+		free(src_reg);
+		rz_list_free(args);
+		return false;
+	}
 
-			value_dst = rz_reg_get_value(core->analysis->reg, item_dst);
-			rz_reg_arena_swap(core->analysis->reg, false);
-			diff_dst = rz_reg_get_value(core->analysis->reg, item_dst);
-			rz_reg_arena_swap(core->analysis->reg, false);
-			rz_list_foreach (head, iter_src, item_src) {
-				ut64 diff_src, value_src;
-				if (!rz_list_find(reg_read, item_src->name,
-					    (RzListComparator)strcmp, NULL)) {
-					continue;
-				}
-				// you never mov from flags
-				if (item_src == item_dst || isFlag(item_src)) {
-					continue;
-				}
-				value_src = rz_reg_get_value(core->analysis->reg, item_src);
-				rz_reg_arena_swap(core->analysis->reg, false);
-				diff_src = rz_reg_get_value(core->analysis->reg, item_src);
-				rz_reg_arena_swap(core->analysis->reg, false);
-				// restore initial value
-				rz_reg_set_value(core->analysis->reg, item_src, diff_src);
-				if (value_dst == value_src && value_dst != diff_dst) {
-					mov = rz_str_appendf(mov, "%s <-- %s;",
-						item_dst->name, item_src->name);
-				}
-			}
-		}
-	continue_error:
-		FREE_ROP;
+	rop_constraint->type = MOV_OP_CONST;
+	rop_constraint->args[DST_REG] = dst_reg;
+	rop_constraint->args[SRC_REG] = src_reg;
+	RzILOpPureCode *op = rz_list_get_n(args, 0);
+	if (!op) {
+		free(dst_reg);
+		free(src_reg);
+		rz_list_free(args);
+		return false;
 	}
-	return mov;
-out_error:
-	FREE_ROP;
-	return NULL;
+
+	char op_str[16];
+	snprintf(op_str, sizeof(op_str), "%s", rz_il_op_pure_code_stringify(*op));
+	rop_constraint->args[OP] = strdup(op_str);
```

Comment:
there is no need to copy the output of `rz_il_op_pure_code_stringify` twice. just

```
const char *op_str = rz_il_op_pure_code_stringify(*op);
rop_constraint->args[OP] = rz_str_dup(op_str);
```